keps/sig-node/4569-cgroup-v1-maintenance-mode/README.md

```diff
@@ -0,0 +1,657 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Move cgroup v1 support in Kubernetes into maintenance mode, aligning with the industry's move towards cgroup v2 as the default for Linux kernel resource management and isolation.
+
+## Motivation
+
+The Linux kernel community has made cgroup v2 the focus for new features, offering better functionality, a more consistent interface, and improved scalability. As a result, major Linux distributions and projects like systemd are phasing out support for cgroup v1. This trend puts pressure on Kubernetes to follow suit.
+
+By shifting cgroup v1 support to maintenance mode, Kubernetes can stay in line with these changes, ensuring compatibility and taking advantage of the improvements in cgroup v2. This transition encourages the use of a more secure and efficient technology while acknowledging that the broader ecosystem, including essential components like the Linux kernel and systemd, is moving beyond cgroup v1.
+
+For those needing long-term support for cgroup v1, it's important to note that this KEP reflects a broader shift in the ecosystem. The reality is that many critical dependencies are moving to cgroup v2, making it necessary for Kubernetes to adapt accordingly.
+
+### Goals
+
+1. **Feature Freeze**: No new features will be added to the cgroup v1 support code. The existing functionality of cgroup v1 will be considered complete and stable.
+
+2. **e2e Testing**: Maintain a set of e2e tests to ensure ongoing validation of cgroup v1 for the currently supported features
+
+3. **Security Maintenance**: Kubernetes community may provide security fixes for Critical and Important CVEs related to cgroup v1  as long as the release is not in end of life.
+
+4. **Best-Effort Bug Fixes**:
+   - Address critical security vulnerabilities in cgroup v1 on priority.
+   - Major bugs in cgroup v1 will be evaluated and potentially fixed if a feasible solution exists.
+   - Acknowledging that some bugs, particularly those requiring substantial changes, may not be resolvable given the constraints around maintaining cgroup v1 support, some issues may need fixes in the kernel or other dependencies that may not happen and so will not be fixed.
+
+5. **Migration Support**: Provide clear and practical migration guidance for users using cgroup v1, facilitating a smoother transition to cgroup v2.
+
+6. **Enhancing cgroup v2 Support**: Address all known pending bugs in Kubernetes’ cgroup v2 support to ensure it reaches a level of reliability and functionality that encourages users to transition from cgroup v1.
```

Comment:
I don't see a direct discussion of this.  Have we met this criteria?  Can we get a link to the cgroups v2 test grid?

Reply:
I added that line to address the concern raised in this comment https://github.com/kubernetes/enhancements/pull/4572#discussion_r1554397743

Reply:
Most likely that particular issue may not be directly related to the k8s, https://github.com/opencontainers/runc/pull/3933#issuecomment-1771110499

Reply:
I suggest working sig-node leads to be sure all known problems have been resolved and shipped in a release to ensure all the fixes work prior to switching to maintenance mode for cgroups v1.